src/msg/async/AsyncConnection.cc

```diff
@@ -130,6 +131,8 @@ AsyncConnection::AsyncConnection(CephContext *cct, AsyncMessenger *m, DispatchQu
   recv_buf = new char[2*recv_max_prefetch];
   if (local) {
     protocol = std::unique_ptr<Protocol>(new LoopbackProtocolV1(this));
+  } else if (m2) {
+    protocol = std::unique_ptr<Protocol>(new ProtocolV2(this));
```

Comment:
It's perhaps a minor detail, but you'll save yourself a potential extra copy by using std::make_unique<> rather than unique_ptr<>'s ctor from the unnamed temporary constructed by operator new.  You can probably get away with: protocol = make_unique<ProtocolV2>(this);